src/Aspire.Hosting.Kafka/KafkaBuilderExtensions.cs

```diff
@@ -110,19 +110,16 @@ public static IResourceBuilder<KafkaServerResource> WithKafkaUI(this IResourceBu
                 .WithHttpEndpoint(targetPort: KafkaUIPort)
                 .ExcludeFromManifest();
 
-            builder.ApplicationBuilder.Eventing.Subscribe<AfterEndpointsAllocatedEvent>((e, ct) =>
+            builder.ApplicationBuilder.Eventing.Subscribe<BeforeResourceStartedEvent>(kafkaUi, (e, ct) =>
```

Comment:
AfterEndpointsAllocatedEvent was being used to queue events before the app model resources were created, but with the changes in the PR for un-proxied container endpoints, it's no longer guaranteed to fire synchronously before resources are created. Using the BeforeResourceStartedEvent works as a replacement.